controllers/openshift_controller.go

```diff
@@ -780,11 +780,22 @@ func (r *KataConfigOpenShiftReconciler) processKataConfigInstallRequest() (ctrl.
 		}
 
 		// Update node selector in machine config pool with value from kataconfig instance
-		r.Log.Info("Updating machine config pool")
+		// When user deletes kataConfigPoolSelector, reset matchLabels in mcp to our defaults
+		// node-role.kubernetes.io/worker and kata-oc
+		r.Log.Info("Updating machine config pool name ", "found Mcp name", foundMcp.Name)
 		if foundMcp != nil {
-			if foundMcp.Spec.NodeSelector != nil {
-				foundMcp.Spec.NodeSelector = r.kataConfig.Spec.KataConfigPoolSelector.DeepCopy()
+			if r.kataConfig.Spec.KataConfigPoolSelector != nil {
+				for key,val := range r.kataConfig.Spec.KataConfigPoolSelector.MatchLabels {
+					foundMcp.Spec.NodeSelector.MatchLabels[key] = val
+				}
```

Comment:
@jensfr What happens if `foundMcp.Spec.NodeSelector.MatchLabels` already contains a key that's not in the current`r.kataConfig.Spec.KataConfigPoolSelector.MatchLabels`, don't we end up with a stale key still hanging around? Shouldn't we clear `foundMcp.Spec.NodeSelector.MatchLabels` first, like we do in the other branch (IOW shouldn't we move the line 793 above the branching on line 787?)?

Reply:
@pmores yes, I think you're right. I think we can get rid of the else branch and always create a new MatchLabels, initialize from scratch and if KataConfigPoolSelector is not empty we add those key-value pairs as well.

Reply:
@jensfr Yeah.  In fact, if we can rely on Kubernetes/MCO to do nothing when applying a resource that hasn't actually changed (I know we can in general but I'm not sure if this still holds even when e.g. the label set remains the same but is reordered) the simplest way might be just to ignore the current contents of `foundMcp.Spec.NodeSelector` altogether and just
- copy `r.kataConfig.Spec.KataConfigPoolSelector.MatchLabels` to `foundMcp.Spec.NodeSelector.MatchLabels`
- copy `r.kataConfig.Spec.KataConfigPoolSelector.MatchExpressions` to `foundMcp.Spec.NodeSelector.MatchExpressions`
- if both `foundMcp.Spec.NodeSelector.MatchLabels` and `foundMcp.Spec.NodeSelector.MatchExpressions` end up empty at this point, run lines 794 and 795 of the current patch.

Am I understanding correctly?